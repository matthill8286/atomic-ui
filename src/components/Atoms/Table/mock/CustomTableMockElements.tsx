import React from 'react'
import { CopyText, Heading } from '@/components/Atoms/Typography'
import { HeadingProps } from '@/components/Atoms/Typography/Heading/Heading.interface'
import { FlexBox, FlexItem, Spacer } from '@/index'
import { styled } from '@/styles/styled'
import { ThemeFontSizes, ThemeFontWeights } from '@/types/theme'
import { InfoText as InfoLine } from '../../Typography/InfoText'

interface TableTitleProps {
  assetName: string
  provider: string
}

interface TableWithDescriptionProps {
  assetWithTitle: TableTitleProps
  description?: string
}

interface TableDataProps {
  asset: TableWithDescriptionProps
  subTexts: string[]
}

const headingProps: HeadingProps = {
  scale: 'level-5',
  tag: 'p',
  color: 'black',
  lineHeight: 'sm',
  fontFamily: 'default',
}

export const StyledTableDescription = styled(CopyText)`
  margin-left: 12px;
`

export const StyledTableDataWrapper = styled(FlexItem)`
  margin-left: 12px;
`

export const TableTitle: React.FC<TableTitleProps> = ({ assetName, provider }) => {
  return (
    <>
      <Heading {...headingProps} margin="0" limitLines={2} data-test="asset-title">
        <CopyText fontSize="sm" tag="span" bold data-test="asset-provider">
          {provider}
        </CopyText>
        {assetName.replace(provider || '', '')}
      </Heading>
    </>
  )
}

export const TableWithDescription: React.FC<TableWithDescriptionProps> = ({
  assetWithTitle,
  description,
}) => {
  return (
    <>
      <TableTitle {...assetWithTitle} />
      {description && (
        <FlexBox flexDirection="row">
          <StyledTableDescription fontSize="sm" tag="span" data-test="asset-description">
            {description}
          </StyledTableDescription>
        </FlexBox>
      )}
    </>
  )
}

export const TableData: React.FC<TableDataProps> = ({ asset, subTexts }) => {
  return (
    <>
      <TableWithDescription {...asset} />
      <Spacer size="xs" />
      {asset.description ? (
        <FlexBox flexDirection="row">
          <StyledTableDataWrapper>
            {subTexts.map((subText, index) => (
              <CopyText key={index} fontSize="xxs" limitLines={0} margin="" tag="p">
                {subText}
              </CopyText>
            ))}
          </StyledTableDataWrapper>
        </FlexBox>
      ) : (
        subTexts.map((subText, index) => (
          <CopyText key={index} fontSize="xxs" limitLines={0} margin="" tag="p">
            {subText}
          </CopyText>
        ))
      )}
    </>
  )
}

interface CustomTextProps {
  text: string
  textWeight?: ThemeFontWeights
  fontSize?: ThemeFontSizes
  additionalInfo?: string
  additionalInfoTextAlign?: 'left' | 'right'
}

export const CustomText: React.FC<CustomTextProps> = ({
  text,
  textWeight = 'semibold',
  fontSize = 'xs',
  additionalInfo,
  additionalInfoTextAlign = 'right',
}) => {
  return (
    <>
      <Heading tag="h1" scale="level-2" weight={textWeight} fontSize={fontSize}>
        {text}
      </Heading>
      {additionalInfo && (
        <InfoLine fontSize={'xs'} textAlign={additionalInfoTextAlign}>
          {additionalInfo}
        </InfoLine>
      )}
    </>
  )
}
