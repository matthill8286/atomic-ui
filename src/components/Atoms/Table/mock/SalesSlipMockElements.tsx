import React from 'react'
import { CopyText, Heading } from '@/components/Atoms/Typography'
import { HeadingProps } from '@/components/Atoms/Typography/Heading/Heading.interface'
import { FlexBox, FlexItem, Spacer } from '@/index'
import { styled } from '@/styles/styled'
import { ThemeFontSizes, ThemeFontWeights } from '@/types/theme'
import { InfoText as InfoLine } from '../../Typography/InfoText'

interface ProductTitleProps {
  productName: string
  manufacturer: string
}

interface ProductWithDescriptionProps {
  productWithTitle: ProductTitleProps
  description?: string
}

interface ProductDataProps {
  product: ProductWithDescriptionProps
  subTexts: string[]
}

const headingProps: HeadingProps = {
  scale: 'level-5',
  tag: 'p',
  color: 'black',
  lineHeight: 'sm',
  fontFamily: 'default',
}

export const StyledProductDescription = styled(CopyText)`
  margin-left: 12px;
`

export const StyledProductDataWrapper = styled(FlexItem)`
  margin-left: 12px;
`

export const ProductTitle: React.FC<ProductTitleProps> = ({ productName, manufacturer }) => {
  return (
    <>
      <Heading {...headingProps} margin="0" limitLines={2} data-test="product-title">
        <CopyText fontSize="sm" tag="span" bold data-test="product-manufacturer">
          {manufacturer}
        </CopyText>
        {productName.replace(manufacturer || '', '')}
      </Heading>
    </>
  )
}

export const ProductWithDescription: React.FC<ProductWithDescriptionProps> = ({
  productWithTitle,
  description,
}) => {
  return (
    <>
      <ProductTitle {...productWithTitle} />
      {description && (
        <FlexBox flexDirection="row">
          <StyledProductDescription fontSize="sm" tag="span" data-test="product-description">
            {description}
          </StyledProductDescription>
        </FlexBox>
      )}
    </>
  )
}

export const ProductData: React.FC<ProductDataProps> = ({ product, subTexts }) => {
  return (
    <>
      <ProductWithDescription {...product} />
      <Spacer size="xs" />
      {product.description ? (
        <FlexBox flexDirection="row">
          <StyledProductDataWrapper>
            {subTexts.map((subText, index) => (
              <CopyText key={index} fontSize="xxs" limitLines={0} margin="" tag="p">
                {subText}
              </CopyText>
            ))}
          </StyledProductDataWrapper>
        </FlexBox>
      ) : (
        subTexts.map((subText, index) => (
          <CopyText key={index} fontSize="xxs" limitLines={0} margin="" tag="p">
            {subText}
          </CopyText>
        ))
      )}
    </>
  )
}

interface PriceTextProps {
  text: string
  textWeight?: ThemeFontWeights
  fontSize?: ThemeFontSizes
  additionalInfo?: string
  additionalInfoTextAlign?: 'left' | 'right'
}

export const PriceText: React.FC<PriceTextProps> = ({
  text,
  textWeight = 'semibold',
  fontSize = 'xs',
  additionalInfo,
  additionalInfoTextAlign = 'right',
}) => {
  return (
    <>
      <Heading tag="h1" scale="level-2" weight={textWeight} fontSize={fontSize}>
        {text}
      </Heading>
      {additionalInfo && (
        <InfoLine fontSize={'xs'} textAlign={additionalInfoTextAlign}>
          {additionalInfo}
        </InfoLine>
      )}
    </>
  )
}
