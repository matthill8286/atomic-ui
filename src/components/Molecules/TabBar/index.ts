export * from './TabBar'
export * from './TabBar.interface'
export * from './TabItem'
