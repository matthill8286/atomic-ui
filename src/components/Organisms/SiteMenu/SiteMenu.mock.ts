export const SiteMenuItems = [
  {
    title: 'Ratings',
    itemId: 'ratings',
  },
  {
    title: 'Technical',
    itemId: 'technical_features',
  },
  {
    title: 'Products',
    itemId: 'product',
  },
]
