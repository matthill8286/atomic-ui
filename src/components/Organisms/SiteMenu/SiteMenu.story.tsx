import { action } from '@storybook/addon-actions'
import { number, select } from '@storybook/addon-knobs'
import { storiesOf } from '@storybook/react'
import * as React from 'react'
import { Button } from '@/components/Atoms/Button'
import { Heading } from '@/components/Atoms/Typography'
import { SiteMenu } from './SiteMenu'
import { SiteMenuItems } from './SiteMenu.mock'

const containerCSS = {
  height: '80vh',
  backgroundColor: 'silver',
  marginBottom: 30,
}

const itemIds = SiteMenuItems.map(i => i.itemId)

storiesOf('Design System/Organisms/SiteMenu', module)
  .add('Default', () => (
    <>
      <SiteMenu
        menuItems={SiteMenuItems}
        surfaceColor="white"
        scrollOffset={number('Scroll Offset', 140)}
        onTracking={action('tracking')}
        activeItemId={select('Active Item', ['None', ...itemIds], 'None')}
      />
      {SiteMenuItems.map((item, index) => (
        <div id={item.itemId} style={containerCSS} key={item.title + index}>
          <Heading scale="level-1" margin="sm xs">
            {item.title}
          </Heading>
        </div>
      ))}
    </>
  ))

  .add('Active Item', () => (
    <>
      <SiteMenu
        menuItems={SiteMenuItems}
        scrollOffset={140}
        surfaceColor="white"
        onTracking={action('tracking')}
        activeItemId="ratings"
      />
      {SiteMenuItems.map((item, index) => (
        <div id={item.itemId} style={containerCSS} key={item.title + index}>
          <Heading scale="level-1" margin="sm xs">
            {item.title}
          </Heading>
        </div>
      ))}
    </>
  ))

  .add('With Child Element', () => (
    <>
      <SiteMenu
        menuItems={SiteMenuItems}
        scrollOffset={140}
        surfaceColor="white"
        onTracking={action('tracking')}
        activeItemId="ratings">
        <Button style={{ alignSelf: 'center', whiteSpace: 'nowrap' }} size="sm">
          I do things
        </Button>
      </SiteMenu>
      {SiteMenuItems.map((item, index) => (
        <div id={item.itemId} style={containerCSS} key={item.title + index}>
          <Heading scale="level-1" margin="sm xs">
            {item.title}
          </Heading>
        </div>
      ))}
    </>
  ))
