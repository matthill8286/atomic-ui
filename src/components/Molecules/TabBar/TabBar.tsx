import React, { FC, Ref, useEffect, useRef, useState } from 'react'
import { Icon } from '@/components/Atoms/Icon'
import { useWindowDimensions } from '@/components/Helper/useWindowDimensions'
import { StyleguideArrow } from '@excelwithbusiness/webmobile-svg-library'
import { TabBarProps, TabData, TabId } from './TabBar.interface'
import {
  StyledArrowButton,
  StyledTabBar,
  StyledTabsContainer,
  StyledTabBarCard,
} from './TabBar.styled'
import { TabItem } from './TabItem'

export const TabBar: FC<TabBarProps> = props => {
  const {
    onChange,
    selected,
    tabs,
    tabGroupName,
    enableSemanticTheme = false,
    surfaceColor = 'surfaceColor',
    elevation = 0,
    noBorder = false,
    isFixed = false,
    changeDelay = 1000,
    flexDirection,
    ...otherProps
  } = props

  const [tab, setTab] = useState(selected || null)
  const [hasOverflow, setHasOverflow] = useState(false)
  const ref: Ref<HTMLDivElement> = useRef<HTMLDivElement>(null)
  const { width } = useWindowDimensions()

  const handleTabClick = (id: TabId, data?: TabData) => {
    setTab(id)
    if (onChange) {
      onChange(id, data)
    }
  }

  useEffect(() => {
    const handler = setTimeout(() => {
      if (tab !== selected && selected) {
        setTab(selected)
      }
    }, changeDelay)

    return () => {
      clearTimeout(handler)
    }
  }, [selected, changeDelay])

  useEffect(() => {
    if (ref.current) {
      const { offsetWidth, scrollWidth } = ref.current
      const hasWidthOverflow = offsetWidth < scrollWidth

      if (hasOverflow !== hasWidthOverflow) {
        setHasOverflow(hasWidthOverflow)
      }
    }
  }, [tabs, width])

  const handlePrevClick = () => {
    if (ref.current) {
      ref.current.scrollBy({
        behavior: 'smooth',
        left: -200,
      })
    }
  }

  const handleNextClick = () => {
    if (ref.current) {
      ref.current.scrollBy({
        behavior: 'smooth',
        left: 200,
      })
    }
  }

  return (
    <StyledTabBar isFixed={isFixed}>
      <StyledTabBarCard
        elevation={elevation}
        shape="rounded-small"
        noBorder={noBorder ? ['top', 'right', 'bottom', 'left'] : []}
        display="flex"
        surfaceColor={surfaceColor}
        enableSemanticTheme={enableSemanticTheme}
        {...otherProps}>
        {hasOverflow && (
          <StyledArrowButton
            actionType="ghost"
            onClick={handlePrevClick}
            surfaceColor={surfaceColor}
            enableSemanticTheme={enableSemanticTheme}>
            <Icon height={16} width={16} rotate={180}>
              <StyleguideArrow />
            </Icon>
          </StyledArrowButton>
        )}
        <StyledTabsContainer flexDirection={flexDirection} ref={ref}>
          {tabs.map(({ id, label, data, isDisabled = false }) => (
            <TabItem
              key={id}
              id={id}
              label={label}
              data={data}
              tabGroupName={tabGroupName}
              isSelected={tab === id}
              isDisabled={isDisabled}
              onClick={handleTabClick}
              enableSemanticTheme={enableSemanticTheme}
              surfaceColor={surfaceColor}
            />
          ))}
        </StyledTabsContainer>
        {hasOverflow && (
          <StyledArrowButton
            actionType="ghost"
            onClick={handleNextClick}
            enableSemanticTheme={enableSemanticTheme}
            surfaceColor={surfaceColor}>
            <Icon height={16} width={16}>
              <StyleguideArrow />
            </Icon>
          </StyledArrowButton>
        )}
      </StyledTabBarCard>
    </StyledTabBar>
  )
}

TabBar.displayName = 'TabBar'
