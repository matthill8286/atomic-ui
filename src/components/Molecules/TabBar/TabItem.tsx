import React, { FC, Ref, useRef, MouseEvent } from 'react'
import { CopyText } from '@/components/Atoms/Typography'
import { TabItemProps } from './TabBar.interface'
import { StyledTabBarItemContainer } from './TabBar.styled'

export const TabItem: FC<TabItemProps> = props => {
  const {
    id,
    label,
    data,
    className,
    tabGroupName,
    enableSemanticTheme = false,
    surfaceColor = 'surfaceColor',
    isDisabled = false,
    isSelected = false,
    onClick,
  } = props
  const ref: Ref<HTMLDivElement> = useRef(null)

  const handleTabClick = (e: MouseEvent) => {
    e.preventDefault()
    if (!isDisabled && onClick) {
      onClick(id, data)
    }
  }

  if (isSelected && ref.current) {
    const rect = ref.current.getBoundingClientRect()
    if (rect.right < 0 || rect.left > window.innerWidth) {
      ref.current.scrollIntoView({ behavior: 'smooth', inline: 'center', block: 'nearest' })
    }
  }

  return (
    <StyledTabBarItemContainer
      onClick={handleTabClick}
      ref={ref}
      className={className}
      surfaceColor={surfaceColor}
      enableSemanticTheme={enableSemanticTheme}>
      <input
        id={id}
        name={tabGroupName}
        type="radio"
        checked={isSelected}
        disabled={isDisabled}
        readOnly
      />
      <CopyText tag="label" htmlFor={id} disabled={isDisabled}>
        {label}
      </CopyText>
    </StyledTabBarItemContainer>
  )
}

TabItem.displayName = 'TabItem'
