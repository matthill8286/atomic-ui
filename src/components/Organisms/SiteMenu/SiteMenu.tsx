import React from 'react'
import { Card } from '@/components/Atoms/Card'
import { Grid } from '@/components/Helper/Grid'
import { TabBar } from '@/components/Molecules/TabBar'
import { Tab } from '@/components/Molecules/TabBar/TabBar.interface'
import { StyledCard } from '@/components/Atoms/Card/Card.styled'
import { styled, css } from '@/styles/styled'
import { Elevation, ThemeColors } from '@/types/theme'
import { smoothScrollWithSelector } from '@/utils'

export interface SiteMenuItemProps {
  itemId: string
  title: string
  surfaceColor?: ThemeColors
}

const StyledGrid = styled(Grid)`
  display: flex;

  > :first-child {
    flex-grow: 1;
    overflow: hidden;
  }
`

const StyledSiteMenuCard = styled(Card)<{
  enableSemanticTheme: boolean
  surfaceColor: ThemeColors
}>`
  position: sticky;
  top: 0;
  z-index: 2;

  ${StyledCard} {
    ${({ enableSemanticTheme, theme, surfaceColor }) =>
      enableSemanticTheme &&
      css`
        background-color: ${theme.color[surfaceColor]};
        color: ${theme.color.textColor};
      `}
  }
`

export interface SiteMenuProps {
  enableSemanticTheme?: boolean
  surfaceColor?: ThemeColors
  isFixed?: boolean
  menuItems?: SiteMenuItemProps[]
  noBorder?: boolean
  elevation?: Elevation
  activeItemId?: string
  scrollOffset?: number
  changeDelay?: number
  onTracking?: (e: string) => void
  onChange?: (id: string, title: string) => void
}

export const SiteMenu: React.FC<SiteMenuProps> = React.memo(
  ({
    isFixed,
    menuItems,
    activeItemId,
    scrollOffset = 0,
    onTracking,
    onChange,
    changeDelay,
    enableSemanticTheme = false,
    children,
    surfaceColor = 'surfaceColor',
  }) => {
    if (!menuItems || !menuItems.length) {
      return null
    }

    const tabs: Tab[] = menuItems.map(item => ({
      label: item.title,
      id: `tab-${item.itemId}`,
      data: { anchor: item.itemId },
    }))

    const handleChange = (id, data) => {
      smoothScrollWithSelector(`#${data?.anchor}`, scrollOffset)

      const item = menuItems.find(i => i.itemId === data.anchor)

      if (onTracking) {
        onTracking(item?.title || '')
      }

      if (onChange) {
        onChange(item?.itemId || '', item?.title || '')
      }
    }

    return menuItems && menuItems.length > 0 ? (
      <StyledSiteMenuCard
        elevation={2}
        shape="sharp"
        enableSemanticTheme={enableSemanticTheme}
        surfaceColor={surfaceColor}>
        <StyledGrid>
          <TabBar
            isFixed={isFixed}
            tabs={tabs}
            tabGroupName=""
            noBorder
            elevation={0}
            onChange={handleChange}
            selected={activeItemId ? `tab-${activeItemId}` : undefined}
            changeDelay={changeDelay}
            enableSemanticTheme={enableSemanticTheme}
            surfaceColor={surfaceColor}
          />
          {children}
        </StyledGrid>
      </StyledSiteMenuCard>
    ) : null
  }
)
