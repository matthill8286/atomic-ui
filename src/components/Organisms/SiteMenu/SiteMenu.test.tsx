import * as React from 'react'
import { TabItem } from '@/components/Molecules/TabBar/TabItem'
import { mountWithTheme, renderWithTheme } from '@/testRenderer'
import { SiteMenu } from './SiteMenu'
import { SiteMenuItems } from './SiteMenu.mock'

describe('SiteMenu', () => {
  it('matches snapshot', () => {
    const tree = renderWithTheme(<SiteMenu menuItems={SiteMenuItems} />)
    expect(tree).toMatchSnapshot()
  })
  it('renders correct', () => {
    const comp = mountWithTheme(<SiteMenu menuItems={SiteMenuItems} />)
    const rows = comp.find(TabItem)
    expect(rows).toHaveLength(SiteMenuItems.length)
    expect(comp).toMatchSnapshot()
  })

  it('renders nothing when undefined', () => {
    const comp = mountWithTheme(<SiteMenu menuItems={undefined} />)
    const menu = comp.find(SiteMenu)
    expect(menu.children().exists()).toBeFalsy()
    expect(comp).toMatchSnapshot()
  })

  it('renders nothing when empty features', () => {
    const comp = mountWithTheme(<SiteMenu menuItems={[]} />)
    const menu = comp.find(SiteMenu)
    expect(menu.children().exists()).toBeFalsy()
    expect(comp).toMatchSnapshot()
  })
})
