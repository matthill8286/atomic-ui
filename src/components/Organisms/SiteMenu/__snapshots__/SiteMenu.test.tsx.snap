// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteMenu matches snapshot 1`] = `
.c9 {
  color: #666666;
  font-size: 14px;
  line-height: 20px;
  font-family: "Oswald","Raleway",Arial,sans-serif;
  font-weight: 400;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c2 {
  display: block;
  box-sizing: border-box;
  line-height: 1.43;
  overflow: hidden;
  box-shadow: 0 3px 4px 0 rgba(0,0,0,0.07),0 3px 3px -2px rgba(0,0,0,0.06),0 1px 8px 0 rgba(0,0,0,0.10);
  color: #000000;
  background-color: #FFFFFF;
  border-radius: 0;
  height: 100%;
  width: 100%;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  line-height: 1.43;
  overflow: hidden;
  color: #000000;
  border: 1px solid #E5E5E5;
  background-color: #FFFFFF;
  border-top-width: 0px;
  border-right-width: 0px;
  border-bottom-width: 0px;
  border-left-width: 0px;
  border-radius: 4px;
  height: 100%;
  width: 100%;
}

.c0 {
  position: relative;
  outline: none;
  width: 100%;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  white-space: nowrap;
  overflow-x: auto;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-scrollbar-width: none;
  -moz-scrollbar-width: none;
  -ms-scrollbar-width: none;
  scrollbar-width: none;
}

.c7::-webkit-scrollbar {
  display: none;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c8 label {
  color: #999999;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  line-height: 1;
  place-items: center;
  place-content: center;
  font-family: "Oswald","Raleway",Arial,sans-serif;
  font-size: 14px;
  text-transform: uppercase;
  font-weight: 700;
  width: 100%;
  padding: 0 16px;
  height: 100%;
  min-height: 52px;
  position: relative;
  text-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c8 label::after {
  -webkit-animation: dRkaku 0.2s linear forwards;
  animation: dRkaku 0.2s linear forwards;
  background-color: #00C8DC;
  bottom: 0;
  content: ' ';
  display: none;
  height: 2px;
  left: 0;
  position: absolute;
  right: 0;
  width: 100%;
}

.c8 input[type='radio'] {
  display: none;
}

.c8 input[type='radio']:checked ~ label {
  font-weight: 600;
  color: #000000;
}

.c8 input[type='radio']:checked ~ label::after {
  display: block;
}

.c8 input[type='radio']:enabled:hover ~ label::after {
  display: block;
}

.c3 {
  box-sizing: border-box;
  margin: 0 auto;
  max-width: 1472px;
  padding: 0 16px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 >:first-child {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  overflow: hidden;
}

.c1 {
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  z-index: 2;
}

@media (min-width:512px) {

}

@media screen and (-ms-high-contrast:active),screen and (-ms-high-contrast:none) {
  .c5 {
    display: none;
  }
}

@media (min-width:752px) {
  .c8 label {
    font-size: 16px;
  }
}

@media (max-width:752px) {
  .c8 label {
    padding: 0 16px;
  }
}

@media (min-width:512px) {
  .c3 {
    padding: 0 24px;
  }
}

@media (min-width:752px) {
  .c3 {
    padding: 0 32px;
  }
}

@media (min-width:1008px) {
  .c3 {
    padding: 0 48px;
  }
}

@media (min-width:1232px) {
  .c3 {
    padding: 0 64px;
  }
}

<div
  className="c0 c1"
>
  <div
    className="c2"
    display="block"
    elevation={2}
    shape="sharp"
  >
    <div
      className="c3 c4"
    >
      <div
        className="c5"
      >
        <div
          className="c0 "
        >
          <div
            className="c6"
            display="flex"
            elevation={0}
            shape="rounded-small"
          >
            <div
              className="c7"
            >
              <div
                className="c8"
                onClick={[Function]}
              >
                <input
                  checked={false}
                  disabled={false}
                  id="tab-ratings"
                  name=""
                  readOnly={true}
                  type="radio"
                />
                <label
                  className="c9"
                  color="grey5"
                  disabled={false}
                  fontSize="xs"
                  htmlFor="tab-ratings"
                >
                  Ratings
                </label>
              </div>
              <div
                className="c8"
                onClick={[Function]}
              >
                <input
                  checked={false}
                  disabled={false}
                  id="tab-technical_features"
                  name=""
                  readOnly={true}
                  type="radio"
                />
                <label
                  className="c9"
                  color="grey5"
                  disabled={false}
                  fontSize="xs"
                  htmlFor="tab-technical_features"
                >
                  Technical
                </label>
              </div>
              <div
                className="c8"
                onClick={[Function]}
              >
                <input
                  checked={false}
                  disabled={false}
                  id="tab-product"
                  name=""
                  readOnly={true}
                  type="radio"
                />
                <label
                  className="c9"
                  color="grey5"
                  disabled={false}
                  fontSize="xs"
                  htmlFor="tab-product"
                >
                  Products
                </label>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SiteMenu renders correct 1`] = `ReactWrapper {}`;

exports[`SiteMenu renders nothing when empty features 1`] = `ReactWrapper {}`;

exports[`SiteMenu renders nothing when undefined 1`] = `ReactWrapper {}`;
